src/Microsoft.DotNet.Build.Tasks/Targets/publishtest.targets

```diff
@@ -20,18 +21,11 @@
 
     <Exec Command="$(NugetRestoreCommand) &quot;$(TestRuntimePackageConfig)&quot;" StandardOutputImportance="Low" />
 
-    <!-- DNU will emit a project.lock.json.  When multiple tests are building for the first time they may race on 
-         running this target and generating the lock.json.  This is causing IOExceptions in dnu for writing lock.json.
-         We should really be handling these test dependencies in a different way (as a package themselves) but for now
-         just restore the project.json from intermediate to avoid the race.  -->
-    <Copy SourceFiles="$(TestRuntimeProjectJson)" DestinationFolder="$(IntermediateOutputPath)">
-      <Output TaskParameter="CopiedFiles" PropertyName="TempTestRuntimeProjectJson"/>
-    </Copy>
-    <Exec Command="$(DnuRestoreCommand) &quot;$(TempTestRuntimeProjectJson)&quot;" StandardOutputImportance="Low" CustomErrorRegularExpression="^Unable to locate .*" />
+    <ExecWithMutex Command="$(DnuRestoreCommand) &quot;$(TestRuntimeProjectJson)&quot;" MutexName="$(TestRuntimeProjectJson)" StandardOutputImportance="Low" CustomErrorRegularExpression="^Unable to locate .*" />
```

Comment:
This seems like a pretty big hammer do you know if this actually causes any noticeable performance hits? I also assume this is temporary so it maybe worth a comment here about that.